### ISSUE TITLE
Exception thrown: 'System.MissingMethodException' in CsvHelper.Excel.dll

### ISSUE THREAD

**Issue by andres-olivares-ncontracts** [JoshClose/CsvHelper#2280]
I updated from from CsvHelper 27.2.1 to 30.0.1, and when calling the following:

using var stream = new MemoryStream(); await using (var excelWriter = new ExcelWriter(stream, new CultureInfo("en-US"))) // Fails - Exception thrown: 'System.MissingMethodException' in CsvHelper.Excel.dll { await excelWriter.WriteRecordsAsync(collection); }

return stream.ToArray();

![image](https://github.com/user-attachments/assets/fa2d7fec-471b-41d8-8594-9107dfacf7f8)

![image](https://github.com/user-attachments/assets/4032bbc3-674a-4391-9206-3277bedefc5d)

Reverted back to CsvHelper 27.2.1 to keep it moving.

Originally mentioned [here](https://github.com/JoshClose/CsvHelper/issues/2082).

Thanks so much in advance,

**Comment by JoshClose** [comment]
You'll need to submit an issue to that library.

**Comment by andres-olivares-ncontracts** [comment]
What are the steps to do that? I thought I was doing that since it is occurring in the latest version of CsvHelper. If you mean CsvHelper.Excel and it has its own place to do this, how would I get there?

**Comment by JoshClose** [comment]
Yes, sorry. That library uses this one and needs to be updated to match changes in this one. It looks like they were matching the numbering system and haven't updated it in a while.

**Comment by andres-olivares-ncontracts** [comment]
I requested an update for this issue, but meanwhile how are you guiding your customers on upgrades when they have this issue to contend with? Has the current practice been to not upgrade, or is there a fix I am unaware of that allows to proceed using the latest CsvHelper? Just want to make sure I am not missing anything.

**Comment by JoshClose** [comment]
I have nothing to do with that Excel library. Personally, I would fix the issue in that library and submit a pull request. It looks like it's not maintained anymore. It also looks like this was forked from another library that isn't maintained anymore. You would have to fork it and make the changes. Honestly, I would say to find another library to migrate to at some point, unless someone else has forked and taken over.

Unless there is a specific feature of CsvHelper that you want, there's no need to update it to the latest. If you're only using it because it's a dependency of the Excel library, then you for sure don't need to update it. If you're using CsvHelper and the Excel library, that's where you may run into a problem.

For the most part, I don't even know about dependencies of CsvHelper. They don't contact me.

**Comment by jdanielp** [comment]
@JoshClose Did you mean "CsvHelper.Excel" rather than "CsvHelper"?

> For the most part, I don't even know about dependencies of **CsvHelper\***. They don't contact me.

\* emphasis mine

Sorry for splitting hairs, just making sure I understand :)
